### PR TITLE
Add Parameter `media_write_timeout` to `HTTPXRequest` and Method `ApplicationBuilder.media_write_timeout`

### LINKED ISSUE #3864: Review types of optional arguments in methods and classes
There are certain parameters in our bot methods which optional with the default as `DEFAULT_NONE` (as specified by the `Defaults` class). The type hinting of such parameters is not consistent. For e.g. 
https://github.com/python-telegram-bot/python-telegram-bot/blob/8cc55c338161e71a640b24a382a259cff6e7a715/telegram/_bot.py#L7197
and 
https://github.com/python-telegram-bot/python-telegram-bot/blob/8cc55c338161e71a640b24a382a259cff6e7a715/telegram/_bot.py#L7201

both parameters accept `bool`, but one of them is type hinted to also accept `None` (ODV -> Optional DefaultValue). Here the type hint for `protect_content` should be corrected to `DVInput` imo.

We should look for other cases like this, and correct it in both the type hints and documentation.

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Last part of and closes #3864

### CODE REVIEW

File: telegram/ext/_applicationbuilder.py [hunk 1]
```diff
@@ -617,6 +628,26 @@ def write_timeout(self: BuilderType, write_timeout: Optional[float]) -> BuilderT
         self._write_timeout = write_timeout
         return self
 
+    def media_write_timeout(
+        self: BuilderType, media_write_timeout: Optional[float]
+    ) -> BuilderType:
+        """Sets the media write operation timeout for the
```
Comment (harshil21) [inline telegram/ext/_applicationbuilder.py:634]: do we want to document the precedence order of choosing the write timeout? i.e. the code block in https://github.com/python-telegram-bot/python-telegram-bot/issues/3864#issuecomment-1712796820. Specifically, we should document that changing `AppBuilder.write_timeout(x).build()` does NOT have an effect on the write timeout for uploading files, i.e. that stays 20 secs by default.

File: telegram/request/_httpxrequest.py [hunk 2]
```diff
@@ -142,6 +149,7 @@ def __init__(
             )
 
         self._http_version = http_version
+        self._media_write_timeout = media_write_timeout
```
Comment (harshil21) [inline telegram/request/_httpxrequest.py:152]: don't we need an entry for this in the `__slots__`? 

The tests once again didn't pick it up cause of the same reason in #4130 ..

**Review by harshil21 — CHANGES_REQUESTED** [review]
implementation looks good. found one place we can improve the documentation and a bug.

File: telegram/ext/_applicationbuilder.py (hunk 1 above)
Comment (Bibo-Joshi) [reply telegram/ext/_applicationbuilder.py:634]: The correct place for elaborating the documentation would be `HTTPXRequest` IMO. All the methods in AppBuilder joust point to the respective docs. I'll add a note to `HTTPXRequest.param.write_timeout` pointing out that it doesn't affect files.

**Review by harshil21 — APPROVED** [review]

**Review by clot27 — APPROVED** [review]

**Pull request merged**